middleware/jwt/authorization.js

```diff
@@ -0,0 +1,73 @@
+const authorizeAdmin = (req, res, next) => {
+  if (req.user.role === 'admin') {
```

Comment:
Wouldn't it be better to extract the role from the token, to avoid multiple redundant queries?

Reply:
I think yes, it is better to get the role from the token.

Reply:
In the token verification, if the token is valid, we add in the request the user from the token. We can access req.user.role